vyper/semantics/types/module.py

```diff
@@ -116,7 +117,8 @@ def _is_function_implemented(fn_name, fn_type):
                 return False
 
             to_compare = fns_by_name[fn_name]
-            assert isinstance(to_compare, ContractFunctionT)
+            assert to_compare.is_external
+            assert all(isinstance(f, ContractFunctionT) for f in [to_compare, fn_type])
```

Comment:
nit:
```suggestion
            assert isinstance(to_compare, ContractFunctionT)
            assert isinstance(fn_type, ContractFunctionT)
```

Reply:
addressed in: https://github.com/vyperlang/vyper/pull/4178/commits/878b966b8133c47f80a2e8c894512e929fdd75a1